controllers/applicationsnapshotenvironmentbinding_controller.go

```diff
@@ -0,0 +1,241 @@
+	if err != nil {
+		log.Error(err, "unable to create temp directory for gitops resources due to error")
+		r.SetCreateConditionAndUpdateCR(ctx, &appSnapshotEnvBinding, err)
+		return ctrl.Result{}, fmt.Errorf("unable to create temp directory for gitops resources due to error: %v", err)
+	}
+
+	// Get the Snapshot CR
+	appSnapshot := appstudioshared.ApplicationSnapshot{}
+	err = r.Get(ctx, types.NamespacedName{Name: snapshotName, Namespace: appSnapshotEnvBinding.Namespace}, &appSnapshot)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("unable to get the Application Snapshot %s %v", snapshotName, req.NamespacedName))
+		r.SetCreateConditionAndUpdateCR(ctx, &appSnapshotEnvBinding, err)
+		return ctrl.Result{}, nil
+	}
+
+	if appSnapshot.Spec.Application != applicationName {
+		err := fmt.Errorf("application snapshot %s does not belong to the application %s", snapshotName, applicationName)
+		log.Error(err, "")
+		r.SetCreateConditionAndUpdateCR(ctx, &appSnapshotEnvBinding, err)
+		return ctrl.Result{}, nil
+	}
+
+	var remoteURL, gitOpsURL, gitOpsBranch, gitOpsContext string
+	componentGeneratedResources := make(map[string][]string)
+
+	for _, component := range components {
+		componentName := component.Name
+
+		// Get the Component CR
+		hasComponent := appstudiov1alpha1.Component{}
+		err = r.Get(ctx, types.NamespacedName{Name: componentName, Namespace: appSnapshotEnvBinding.Namespace}, &hasComponent)
+		if err != nil {
+			log.Error(err, fmt.Sprintf("unable to get the Component %s %v", componentName, req.NamespacedName))
+			r.SetCreateConditionAndUpdateCR(ctx, &appSnapshotEnvBinding, err)
+			return ctrl.Result{}, nil
+		}
+
+		// Sanity check to make sure the binding component has referenced the correct application
+		if hasComponent.Spec.Application != applicationName {
+			err := fmt.Errorf("component %s does not belong to the application %s", componentName, applicationName)
+			log.Error(err, "")
+			r.SetCreateConditionAndUpdateCR(ctx, &appSnapshotEnvBinding, err)
+			return ctrl.Result{}, nil
+		}
+
+		var clone bool
+		var imageName string
+
+		for _, snapshotComponent := range appSnapshot.Spec.Components {
+			if snapshotComponent.Name == componentName {
+				imageName = snapshotComponent.ContainerImage
+			}
+		}
+
+		if imageName == "" {
+			err := fmt.Errorf("application snapshot %s did not reference component %s", snapshotName, componentName)
+			log.Error(err, "")
+			r.SetCreateConditionAndUpdateCR(ctx, &appSnapshotEnvBinding, err)
+			return ctrl.Result{}, nil
+		}
+
+		gitopsStatus := hasComponent.Status.GitOps
+		// Get the information about the gitops repository from the Component resource
+		gitOpsURL = gitopsStatus.RepositoryURL
+		if gitOpsURL == "" {
+			err := fmt.Errorf("unable to create gitops resource, GitOps Repository not set on component status")
+			log.Error(err, "")
+			r.SetCreateConditionAndUpdateCR(ctx, &appSnapshotEnvBinding, err)
+			return ctrl.Result{}, nil
+		}
+		if gitopsStatus.Branch != "" {
```

Comment:
Should we have some common utils function to handle these default values for Branch/Context? IIRC, we have similar logic in either the Application/Component controller.